plugins/chat/app/models/chat/message.rb

```diff
@@ -91,11 +95,56 @@ class Message < ActiveRecord::Base
 
     before_save { ensure_last_editor_id }
 
+    normalizes :blocks,
+               with: ->(blocks) do
+                 return if !blocks
+
+                 # automatically assigns unique IDs
+                 blocks.each do |block|
+                   block["block_id"] ||= SecureRandom.uuid
+                   block["elements"].each do |element|
+                     element["action_id"] ||= SecureRandom.uuid if element["type"] == "button"
+                   end
+                 end
+               end
+
     validates :cooked, length: { maximum: 20_000 }
-    validate :validate_message
+
+    validate :validate_blocks
+    def validate_blocks
```

Comment:
That validation block seems rather complex, should we extract it to a dedicated validation class?